lib/puppet/util/windows/security.rb

```diff
@@ -328,15 +330,35 @@ def set_mode(mode, path, protected = true)
       # It is possible to set SYSTEM with a mode other than Full Control (7) however this makes no sense and in practical terms
       # should not be done.  We can trap these instances and correct them before being applied.
       if (sd.owner == well_known_system_sid) && (owner_allow != FILE::FILE_ALL_ACCESS)
-        #TRANSLATORS 'SYSTEM' is a Windows name and should not be translated
-        Puppet.debug _("An attempt to set mode %{mode} on item %{path} would result in the owner, SYSTEM, to have less than Full Control rights. This attempt has been corrected to Full Control") % { mode: mode.to_s(8), path: path }
-        owner_allow = FILE::FILE_ALL_ACCESS
+        # If owner and group are both SYSTEM but group is unmanaged the control rights of system will be set to FUllControl by
+        # the unmanaged group, so there is no need for the warning
+        if managing_owner && !(isownergroup && !managing_group)
```

Comment:
`!(isownergroup && !managing_group)` is the same as `(!isownergroup || managing_group)`  Does that read better?

Reply:
Yeah, I think it might read a little better.